docs/doxygen/pages.txt

```diff
@@ -51,7 +51,120 @@ The configurations for memory estimation are defined <a href="https://github.com
+wait for acknowledgments, a call to @ref mqtt_processloop_function or @ref mqtt_receiveloop_function must follow in order to receive any expected acknowledgment.
+The exception is @ref mqtt_connect_function; since a connection cannot be considered established until the server accepts a CONNECT packet with a CONNACK,
+the function waits until the CONNACK is received.
+
+@subsection mqtt_receivetimeout Timeouts
+@ref mqtt_connect_function, @ref mqtt_processloop_function, and @ref mqtt_receiveloop_function all accept timeout parameters for packet reception.<br>
+For the @ref mqtt_connect_function, if this value is set to 0, then instead of a time-based loop, it will attempt to call the transport receive function up
+to a fixed maximum number of times, which is defined by @ref MQTT_MAX_CONNACK_RECEIVE_RETRY_COUNT.
+
+For @ref mqtt_processloop_function and @ref mqtt_receiveloop_function, the timeout value represents the <i>minimum</i> duration that will be spent in the function, provided there are no network errors.
+Should the timeout be set to 0, then the loop will run for a single iteration. A single iteration of a loop consists of an attempt to receive a single byte from the network,
+followed by the reception of the entire packet if the receive was successful, and finally deserialization of the packet received and a call to the application callback. If the first read did not succeed,
+then instead the library checks if a ping request needs to be sent (only for the process loop).
+
+See the below diagrams for a representation of the above flows:
+| MQTT Connect Diagram | MQTT ProcessLoop Diagram | MQTT ReceiveLoop Diagram |
+| :--: | :--: | :--: |
+|<img src="mqtt_connect_design.png" alt="MQTT Connect" width=100% /> |<img src="mqtt_processloop_design.png" alt="MQTT Process Loop" width=100% /> |<img src="mqtt_receiveloop_design.png" alt="MQTT Receive Loop" width=100% /> |
+
+@section mqtt_keepalive Keep-Alive
+
+The MQTT standard specifies a keep-alive mechanism to detect half-open or otherwise unusable network connections.
+An MQTT client will send periodic ping requests (PINGREQ) to the server if the connection is idle. The MQTT server must respond to ping requests with a ping response (PINGRESP).
+
+In this library, @ref mqtt_processloop_function handles the sending of PINGREQs and checking of PINGRESPs to comply with the keep-alive interval set in @ref MQTTContext_t.keepAliveIntervalSec.
+
+The standard does not specify how long the server has to respond to a ping request, noting only a "reasonable amount of time". In this library, the amount of time a server has to respond to a ping request is set with @ref MQTT_PINGRESP_TIMEOUT_MS.
+
+If @ref mqtt_receiveloop_function is used instead of @ref mqtt_processloop_function, then no such keep-alive checks are performed, and the application must ensure the keep-alive interval is obeyed by calling @ref mqtt_ping_function whenever control packets have not been sent for some period of time.
```

Comment:
```suggestion
If @ref mqtt_receiveloop_function is used instead of @ref mqtt_processloop_function, then no ping requests are sent. The application must ensure the connection does not remain idle for more than the keep-alive interval by calling @ref mqtt_ping_function to send ping requests.
```